src/stactools/core/utils/raster_footprint.py

```diff
@@ -0,0 +1,296 @@
+            the tolerance distance of the original geometry, in degrees.
+        no_data(Optional[int]): explicitly set the no data value if not in image metadata
+
+    Returns:
+        bool: True if the extent was successfully updated, False if not
+    """
+    asset_name_and_extent = next(
+        data_footprints_for_data_assets(
+            item,
+            asset_names=asset_names,
+            scale=scale,
+            precision=precision,
+            densification_factor=densification_factor,
+            simplify_tolerance=simplify_tolerance,
+            no_data=no_data,
+        ),
+        None,
+    )
+    if asset_name_and_extent is not None:
+        _, extent = asset_name_and_extent
+        item.geometry = extent
+        return True
+    else:
+        return False
+
+
+def data_footprints_for_data_assets(
+    item: Item,
+    *,
+    asset_names: Optional[List[str]] = None,
+    scale: int = DEFAULT_SCALE,
+    precision: int = DEFAULT_PRECISION,
+    densification_factor: Optional[int] = None,
+    simplify_tolerance: Optional[float] = None,
+    no_data: Optional[int] = None,
+) -> Iterator[Tuple[str, Dict[str, Any]]]:
+    """
+    Accepts an Item and an optional list of asset names within that item, and produces
+    an iterator over the same asset names (if they exist and have the role "data") and
+    a dictionary representing a GeoJSON Polygon of the data footprint of the asset.
+    The data footprint is considered to be a convex hull around all areas within the
+    raster that have data values (e.g., they do not have the "no data" value).
+
+    See :py:meth:`update_geometry_from_asset_footprint` for more details about
+    densification and simplification.
+
+    Args:
+        item (Item): The PySTAC Item to extend.
+        asset_names (Optional[List[str]]): The names of the assets for which to extract
+            footprints.
+        scale (int): A factor to scale out the shape to. Defaults to 1.
+        precision (int): The number of decimal places to include in the coordinates for
+            the reprojected geometry. Defaults to 3 decimal places.
+        densification_factor (Optional[int]): The factor by which to increase point density
+            within the polygon. A factor of 2 would double the density of points (placing one
+            new point between each existing pair of points), a factor of 3 would place two points
+            between each point, etc.
+        simplify_tolerance (Optional[float]): All points in the simplified object will be within
+            the tolerance distance of the original geometry, in degrees.
+        no_data(Optional[int]): explicitly set the no data value if not in image metadata
+    Returns:
+        Iterator[Tuple[str, Dict[str, Any]]]: Iterator of the data extent as a geojson dict
+            for each asset.
+    """
+    for name, asset in item.assets.items():
+        if asset.roles and (asset_names is None or name in asset_names):
```

Comment:
What is the reason for the requirement for `roles` to exist on the asset?

Reply:
oh, that's unnecessary now -- previously we were checking that the role was data

Reply:
I'll fix that up, and I also see there's a mention in the documenation to remove